packages/@dcl/inspector/src/components/AssetsCatalog/AssetsCatalog.tsx

```diff
@@ -33,28 +34,30 @@ const AssetsCatalog: React.FC<Props> = ({ catalog }) => {
   )
 
   const filteredCatalog = useMemo(() => {
-    const results = []
+    let results: AssetPack['assets'] = []
+
     if (search) {
-      if (selectedTheme) {
-        for (const asset of selectedTheme.assets) {
-          if (asset.name.toLowerCase().includes(search.toLowerCase())) {
-            results.push(asset)
-          }
-        }
-      } else {
-        for (const theme of catalog) {
-          for (const asset of theme.assets) {
-            if (asset.name.toLowerCase().includes(search.toLowerCase())) {
-              results.push(asset)
-            }
-          }
-        }
+      const assets = selectedTheme ? selectedTheme.assets : catalog.flatMap((theme) => theme.assets)
+      results = assets.filter((asset) => asset.name.toLowerCase().startsWith(search.toLowerCase()))
+
+      if (results.length === 0) {
+        results = assets.filter((asset) => asset.name.toLowerCase().includes(search.toLowerCase()))
       }
     }
 
     return results
   }, [catalog, selectedTheme, search])
 
+  useEffect(() => {
+    if (search) {
+      analytics.track(Event.SEARCH_ITEM, {
```

Comment:
should this be debounced? otherwise it will track every key stroke

Reply:
we alreaady have an implementation: 

https://github.com/decentraland/js-sdk-toolchain/blob/main/packages/%40dcl/inspector/src/components/ui/utils.ts#L5-L11

Reply:
the callback `onSearch => setSearch`: https://github.com/decentraland/js-sdk-toolchain/pull/876/files#diff-44779b63b16cc772140e7e09d5226f757e76b141dc7d6f2db194eca7f5646530R96

is debounced to `500ms` in this component: https://github.com/decentraland/js-sdk-toolchain/pull/876/files#diff-5d19b541c08926fc240ec0c07f2e80323946f4497f23c2205fca7044f148e2d8R57